src/pages/Folder.jsx

```diff
@@ -6,16 +9,55 @@ import FolderToolBar from "../component/FolderToolBar/FolderToolBar";
 import "./page.css";
 
 const Folder = () => {
-  const apiURL = "https://bootcamp-api.codeit.kr/api/users/1";
+  const [folderNameData, setFolderNameData] = useState([]);
+  const [links, setLinks] = useState([]);
+  const [selectedButtonName, setSelectedButtonName] = useState(ALL);
+
+  useEffect(() => {
+    fetchFolders()
+      .then((data) => {
+        setFolderNameData(data);
+        return fetchLinks();
+      })
+      .then((data) => {
+        setLinks(data);
+      })
+      .catch((error) => console.error("Error:", error));
+  }, []);
+
+  const onFolderSelect = (folderId) => {
+    if (folderId === ALL) {
+      setSelectedButtonName(ALL);
+      fetchLinks()
+        .then((data) => {
+          setLinks(data);
+        })
+        .catch((error) => console.error("Error:", error));
+    } else {
+      const folderName =
+        folderNameData.find((button) => button.id === folderId)?.name || ALL;
+      setSelectedButtonName(folderName);
+      fetchLinks(folderId)
+        .then((data) => {
+          setLinks(data);
+        })
+        .catch((error) => console.error("Error:", error));
+    }
+  };
```

Comment:
early return 을 사용해보아요 ㅎㅎㅎ

```suggestion
    if (folderId === ALL) {
      setSelectedButtonName(ALL);
      fetchLinks()
        .then((data) => {
          setLinks(data);
        })
        .catch((error) => console.error("Error:", error));
         return;
    } 
      const folderName =
        folderNameData.find((button) => button.id === folderId)?.name || ALL;
      setSelectedButtonName(folderName);
      fetchLinks(folderId)
        .then((data) => {
          setLinks(data);
        })
        .catch((error) => console.error("Error:", error));
  };
```